usdt-impl/src/record.rs

```diff
@@ -186,32 +199,31 @@ fn process_rec(providers: &mut BTreeMap<String, Provider>, rec: &[u8]) -> Result
     let args = {
         let mut args = Vec::with_capacity(n_args);
         for _ in 0..n_args {
-            args.push(String::from(data.read_cstr()));
+            args.push(limit_string_length(data.read_cstr(), MAX_ARG_TYPE_LEN));
         }
         args
     };
 
     let funcname = match addr_to_info(address).0 {
-        Some(s) => s,
+        Some(s) => limit_string_length(s, MAX_FUNC_NAME_LEN),
         None => format!("?{:#x}", address),
     };
 
-    let provider = providers.entry(provname.to_string()).or_insert(Provider {
-        name: provname.to_string(),
+    let provname = limit_string_length(provname, MAX_PROVIDER_NAME_LEN);
```

Comment:
ok... this is picayune, but if we're going to check this we need to keep in mind that DTrace will append the pid to the process name specified so it's actually MAX_PROVIDER_NAME_LEN - log_10(maxpid) and by default that's 5 or 6

Reply:
Yes of course. And no need to apologize as it's totally opaque and basically impossible for you to have discovered.